### DISCUSSION TITLE
[Tracking] Magic Spells, Curses, and Enchantments

### DISCUSSION THREAD

**Discussion by Sollace** [Sollace/Unicopia#592] [Ideas]
This is the big list of magic suggestions! Got an idea for a spell or mechanic? Reply here with a comment and maybe I'll consider adding it.

Things to consider:
 - Keep it pony related. Either back up by a source from the show, or IDW comics. I only want to add things to help increase the impressiveness of actually _being in Equestria_.
 - Spells can be Good, Bad. Bad spells can typically be more powerful, but they also need to have some drawbacks or negative effects. They are bad, after all.
 - Spells can be thrown. place down at stay in one spot, or travel with the player. Some spells are allowed to function as projectiles. If you have an idea to incorporate that into your suggestion, I'd like to hear.
 - Try to consider how your spell will react to different races. Players can have races, as do items dropped by them.
 - All living mobs and players can have a spell cast on them directly, too.

If you have any questions, you can ask me here or on the Mine Little Pony discord.

Possible reference for ideas:
https://mlp.fandom.com/wiki/Magic

**Comment by Orymizu** [comment]
Perhaps one that transforms the user into a seapony? Or at least grant water breathing/speed.

**Comment by kerplamp** [comment]
I've thought about how the races interact in a minecraft co-op setting and what can be done to encourage team work other than just "Oh we all get blocks and then we all place blocks." like how minecraft normally works. (Horrible simplification btw but I think it gets the point across) and thought I'd comment here. Not sure if this is the right place though. Anyway here are my thoughts on balancing.

Earth ponies could be the resource gatherers, maybe a slight increase to dig speed along-side their current kit would make them perfect for mining. Or maybe the kick ability could destroy a bunch of normal stone when holding a pickaxe for faster building material gathering. High damage to deal with mobs and better mining to get it done faster.

Pegasi are amazing for transport. My friend is a pegasus and we use the ability to pick others up to go look for caves. Pegasi can also help with larger / more vertical builds. Getting blocks to other players, bringing players back to base if the build is far away, or just building parts themselves. Not sure what else pegasi could get added to their kit but my friend has complained a little bit about not being able to go down faster while flying. So maybe a dive ability? I guess  I think just the ability to stop flying mid air would do it though. (Not sure if you can already, I couldn't find any keybinds or key combos when I was messing around with pegasus)

Unicorns were weird to think about because the whole digging and building parts of minecraft are done but I don't want to simply suggest "crafting" because I don't feel like they would be doing as much as the other races. So I thought that instead of making them fill another role, they could just amplify the effects of others. I think a temporary buff to tools or some sort of temporary buff in general would make unicorns mesh better into the process of mining and building. They would have to stay around other players and keep re-applying.

I haven't really thought about combat because it looks like you aced it. Unicorns have a strange kit that makes for a interesting fight, earth ponies are powerhouses at close range but the other races have mobility to counter it, and pegasi have longer reach (I think, I saw it mentioned in the code) and the ability to fly which alone is amazing.

Hope my comment helps with any thoughts on balance or new features. This is a really cool mod with how it expands the minecraft multiplayer dynamic, and I hope to see it updated in the future! :)

**Comment by kerplamp** [comment]
**Enchantments**

**Gem-Finding** (Tool Enchant): Makes a faint noise when valuables are near, getting louder the closer you are. The higher its level the further it can detect from. Level 1: 4 blocks away. Level 2: 6 blocks away, Level 3: 8 blocks away.       [Sound.](https://www.youtube.com/watch?v=-IZ5C82IKtQ)

**Padded** (Armor Enchant): Take less damage from hitting walls and earth ponies' ground pound. Good for clumsy pegasi and can level the playing field against earth ponies' combat abilities. The enchant shouldn't give too much resistance because it can be put on a whole set of armor, stacking the effect.

**Heavy** (Boot enchant): Makes player harder to move. Things like the vortex spell and earth pony ground pound will cause less knockback. Good for pegasi due to their lightweight nature. Could also make earth pony do more damage to blocks when ground pounding. Could also reduce effects of storms on pegasi.

**Harmony** (Weapon Enchant): Weapons with this enchantment will get stronger depending on how many other players with this enchantment are nearby.

**Curses**

**Want It Need It** (Any Item): Passive mobs will attack the player while item is out/equipped . Zombies will have a chance of stealing the item when hitting the player.

**Poison Joke** (Any Item): Plays mob noises around player at random.

**Stress** (Any Item): Screen shakes (Like when you deplete mana as pegasus) more depending on how many enemies surround the player.

**Comment by Sollace** [comment]
@kerplamp The request for spells/enchants/curses was intended as more for Unicopia's own magic system, but these are good ideas.

**Comment by Sollace** [comment]
@kerplamp I have updated this story's description to be more in line with the current state of the mod. Also added a wiki link to various magical artefacts, spells, etc featured in the series, for idea. I'd be interested to hear what you think/can come up with since you're ideas have been very good so far.

**Comment by kerplamp** [comment]
### **Spells**


**Spell of Mending / Restoration Gem, R only.** 
_Reference: [Spell of Relic Reconstituion](https://youtu.be/J6pFdHYaWWE?t=211)_
_R:_ Very slowly repairs items within its circle.

**Spell of Gathering / Harvest Gem, R and V.** 
_Reference: Modified Vortex spell. _
_R_: Breaks fully grown crops in an area around you and brings them to the center of the spell.
_V_: Breaks all crops near the spell and pulls them into the center of the spell.
_Charged V_: Breaks fully grown crops in a smaller radius around the spell and pulls them into the center of the spell.

**Spell of Channeling / Channeling Gem, R and V.** 
_Reference: Staff of Sacanas_
_R_: When raining, increases the chance of lightning striking where the spell is cast. 
_V_: Upon hitting an entity, makes it more likely for that entity to get struck by lightning.
_Charged V_: Will release a short ranged spell that releases lightning where it lands, you will probably hit yourself with this spell, be careful!

**Spell of Signaling / Signaling Gem, R and V.**
_Reference: A changeling uses a flare spell in a fan made animation, [Once upon a time in Canterlot](https://youtu.be/LFjEintbDXc?t=52)_
_R_: Upon using the spell a magical flare will be sent into the sky, signaling for any wandering traders, or pillagers if you're unlucky. Could also just be used to signal friends in case you get lost lol.
_V_: Sets whatever it hits on fire.

**Spell of Banishment / Banishing Gem, R only.**
_Reference: Luna's banishment and lack of moon biome._
_R_: Upon use a circle will spawn in front of the user and begin a long countdown, at the end of the countdown, any entity inside the circle is sent to the nether. Except for like the ender dragon and stuff. 

**Spell of Fertility / Fertile Gem, R only.**
_Reference: No actual reference, just thought it would be useful._
_R_: Upon use, mobs in a radius around the player will have their cooldown before breeding reduced. Or it could breed all mobs in an area regardless of their breeding cooldown. 

**Spell of Illumination I/ Spark Gem, R and V.**
_Reference: Lesser version of light beam spell._
_R_: Upon use you will emit a small amount of light until dispelled.
_V_: Upon use the spell will launch forward and release a bright flash, temporarily blinding any enemies. 
_Charged V_: The spell will travel less distance but have a brighter flash, increasing the length of blindness.

**Spell of Illumination II / Light Gem, R and V.**
_Reference: Generic unicorn light beam spell. Used in fight against Tirek._
_R_: Upon use you will emit a large amount of light until dispelled.
_V_: Shoots out a ball of light that damages and blinds the enemy it hits. 
_Charged V_: Shoot out a ball of light that does a larger amount of damage but does not blind.

**Spell of Trickery / Trick Gem, R and V.**
_Reference: Trixie's illusions in Boast Busters and Magic Duel_
_R_: Creates illusory versions of the blocks you look at. These illusory blocks can be walked through or fallen through.
_V_: Creates smoke where the spell lands.
_Charged V_: Creates an illusory mob that upon either taking damage or attempting to deal damage, disappears into a cloud of smoke.

**Spell of Shadows / Shadow gem, R and V.**
_Reference: Sombra's abilities_
_R_: Upon use the user enters a shadow state where they are immune to melee and ranged damage. Environmental damage like lava and fire are still able to hurt them and will force the user back into their normal form.
_V_: Creates a wall of magic negating crystal in front of the user.
_Charged V_: A spell circle is formed where the spell lands and after a brief cooldown, black, magic negating crystals form around the crystal, trapping anything inside. 



### **Artefacts**

**Amulet of Adoration:** 
_Reference: I don't know if this actually appears in show but it sounds like it would. Some sort of "villain looking for attention" scenario or something._
When wearing this amulet all villagers will reduce prices on their goods. However there is a chance the villager will realize they are being manipulated and will refuse to sell you anything or increase their prices to ludicrous amounts.

**Bunny Hood:**
_Reference: The bunny suit... The results from googling "mlp bunny suit" are not helpful..._
While wearing this item, you’re nametag will be hidden from behind walls and you gain the ability to hide in tall grass from mobs. Or you could get faster sneak speed.

**Music Box /  Stern der Winterträume / Star of Winterdreams:**
_Reference: MLP Wiki, Stern der Winterträume_
When a player sleeps through the night and the music box is within range, the players hunger is refilled.

### **Notes**

**These are very conceptual and I don't really have any exact balance ideas for range, damage, cast time, or duration but I think they are okay ideas for spells.**

I have a bunch of other stuff written down in a google doc but I'm not sure whether I should open a new "issue", post it all here, open multiple "issues" with each themed to a different race, or just send the google doc. 

I guess I'll just send the google doc here anyway so you can see it at least. It's not super detailed cuz its just a bunch of notes and I changed the spells in this post while making it so they no longer match up. 
https://docs.google.com/document/d/1EuW7R8oVPs5MmldReZfqZ9N-sEearhu9pi7toWPTfhs/edit

Hope this helps!
